pkg/generate/app/cmd/newapp.go

```diff
@@ -418,6 +418,19 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 						return nil, err
 					}
 				}
+				// Append any exposed ports from Dockerfile to input image
```

Comment:
> Port detection was previously happening while in source detection (removed by this PR). We should restore it somewhere there and not so late in the new-app flow.

We still get the data during source detection (populated Dockerfile). However, the flow is different now since we do a resolve step which wasn't happening before.